### ISSUE TITLE
Cookies: stricter settings and conflicts with other apps

### ISSUE THREAD

**Issue by unixerius** [Syslifters/sysreptor#271]
Good day! 

I'm running a PoC with both WriteHat and SysReptor, to compare their feature sets, user friendliness, etc. They are both hosted on the same Docker host for the duration of this PoC. In testing I noticed that logging in to one of these would automatically invalidate my current session in the other tool, and vice versa.

After some messing around with BurpSuite and some web searches I'm reminded that this happens in accordance with RFC6265 (HTTP state management).

- Both WriteHat and SysReptor share the same FQDN.
- Both tools use cookies with the names "csrftoken" and "sessionid".
- Due to RFC6265, these cookies are treated as one and the same.

I realise this is a "me" problem, not a "you" problem. Your code operates according to the RFC and your Docker deployment process means the code won't even be aware of the actual FQDN and Origin (proto://fqdn:port) it will be reached upon.

For me, my solution should be to ensure that the NGinx proxy is set up with a separate virtualhost / FQDN for each of the ingresses towards these two tools. 

That being said, there _is_ a feature request I have for you. :)

Would you potentially consider setting the following flags for both your CSRFtoken and your SessionID cookies?
- `SameSite: Strict` because both pieces of information are only relevant for _your_ app.
- `HttpOnly` because they will only be altered by server side responses.
- `Secure` so the are only transferred/accepted across HTTPS.
- `Path: ${yourpath}`, potentially set based upon a new configuration parameter, defaults to `/`.
- `Domain: ${fqdn}`, set based upon an existing configuration parameter.
- `Expires: ${desiredtime}`, so the session persists even if a tab is closed, but it's also not without limitations. Ideally max 4-8h unless renewed during the session.

**Comment by MWedl** [comment]
Hi, we already set some of the mentioned cookie flags and some flags we deliberately do not set.

* `SameSite: Strict`: already set for `sessionid` and `csrftoken`
* `HttpOnly`: Set for `sessionid`. Not set for `csrftoken` because this cookie needs to be readable by JS. CSRF tokens do not require the `HttpOnly` flag to protect against CSRF attacks.
* `Secure`: Currently not set. We will add them. Thanks for noticing.
* `Path`: SysReptor does not support running on a subpath, so we do not set this flag.
* `Domain`: This flag is deliberately unset such that browsers send this cookie only to the domain that set it. Cookies are only valid for a single SysReptor instance. There is no need to send cookies to other domains. 
* `Expires`: This flag is deliberately unset to use session cookies that are discarded when the browser is closed. The backend enforces session timeout of 14 hours.

**Comment by unixerius** [comment]
Thank you for your very swift feedback! Those are very clear statement.

I'm happy that I could at least contribute the suggestion of the `Secure` flag. For the others, I understand why you cannot / will not implement them. 

And thank you for noting that SysReptor does not support running on sub-paths. That's important for us to know! All the more reason to give it its own DNS entry and vhost.

**Comment by aronmolnar** [comment]
Reopening for tracking the resolution.

For `Secure` attribute, we must check if the installation runs via https, because this might not be the case for local installations.

**Comment by MWedl** [comment]
Implemented in https://github.com/Syslifters/sysreptor/releases/tag/2024.49

We introduced the `SECURE_SSL_REDIRECT` setting to configure that SysReptor should only be accessible via https. All http requests will be redirected to https and `Secure` flags will be set for cookies. Reverse proxies should set the HTTP header `X-Forwarded-Proto: https` to signal that incoming requests were TLS-protected https requests.
